app/components/mandatarissen/mandataris/publication-status-selector.js

```diff
@@ -42,18 +58,115 @@ export default class MandatarissenMandatarisPublicationStatusSelectorComponent e
     if (publicationStatus.isBekrachtigd) {
       this.isDisabled = true;
+      this.showLinkToDecisionModal = true;
+    } else {
+      await this.setStatus(publicationStatus);
     }
+  }
+
+  @action
+  async setStatus(publicationStatus) {
     this.mandataris.publicationStatus = publicationStatus;
     await this.mandataris.save();
     if (this.args.onUpdate) {
       this.args.onUpdate();
     }
   }
+
+  @action
+  setType(selectedType) {
+    this.selectedType = selectedType;
+    this.isLinkToPage = selectedType.id === 3;
+  }
+
+  @action
+  setDecisionPredicate(predicate) {
+    this.selectedDecisionPredicate = predicate;
+  }
+
+  @action
+  async cancelAddDecision() {
+    this.selectedPublicationStatus = await this.mandataris.publicationStatus;
+    this.showLinkToDecisionModal = false;
+    this.args.onUpdate();
+  }
+
+  isValidUri(inputValue) {
+    // eslint-disable-next-line no-useless-escape, prettier/prettier
+    const uriRegex = new RegExp(
+      '^https?://[a-zA-Z0-9-]+(.[a-zA-Z0-9-]+)*(/.*)?$'
+    );
+    return uriRegex.test(inputValue);
+  }
+
+  setLinkTodecision = restartableTask(async (event) => {
+    await timeout(INPUT_DEBOUNCE);
+    this.linkToDecision = event.target?.value;
+    this.isInputLinkValid = this.isValidUri(this.linkToDecision);
+  });
+
+  addDecisionToMandataris = task(async () => {
+    if (this.selectedType.id === 3) {
+      this.mandataris.linkToBesluit = this.linkToDecision;
+      await this.mandataris.save();
+      this.showLinkToDecisionModal = false;
+      await this.setStatus(this.selectedPublicationStatus);
+    } else {
+      const searchModel = this.selectedType.id === 1 ? 'artikel' : 'besluit';
+      const rechtsgrondenWithUri = await queryRecord(this.store, searchModel, {
+        filter: {
+          ':uri:': this.linkToDecision,
+        },
+      });
+      if (!rechtsgrondenWithUri) {
+        showWarningToast(
+          this.toaster,
+          'Geen besluit of artikel gevonden voor deze uri',
+          'Geen resultaat'
+        );
+      } else {
+        const rechtsgrondPropertie =
+          this.selectedDecisionPredicate.id === 1
+            ? 'bekrachtigtAanstellingVan'
+            : 'bekrachtigtOntslagVan';
+        rechtsgrondenWithUri[rechtsgrondPropertie] = this.mandataris;
+        await rechtsgrondenWithUri.save();
+        this.showLinkToDecisionModal = false;
+        await this.setStatus(this.selectedPublicationStatus);
+      }
+    }
+  });
+
+  get canSaveLinkToDecision() {
+    if (this.selectedType?.id !== 3) {
+      if (!this.selectedDecisionPredicate) {
+        return false;
+      }
+    }
+
+    return this.isInputLinkValid && this.selectedType;
+  }
+
+  get typeOptions() {
+    return [
+      { id: 1, label: 'Artikel' },
+      { id: 2, label: 'Besluit' },
+      { id: 3, label: 'Link naar pagina' },
+    ];
+  }
+
+  get predicateOptions() {
+    return [
+      { id: 1, label: 'bekrachtigtAanstellingVan' },
```

Comment:
meaning allowing them to select this is also not necessary, since the only thing we care about it the bekrachtiging of the mandataris. The bekrachtiging of the dismissal is less important (it's just the end date for us) and doesn't modify state

Reply:
Removed this